src/main/java/edu/hm/hafner/analysis/IssueFilterConfig.java

```diff
@@ -0,0 +1,279 @@
+ * them there is a possibility to include and exclude issues. The filters are regexes. Instantiation only with
+ * IssueFilterConfigBuilder.
+ */
+public class IssueFilterConfig {
+    private final boolean isIncludeSet;
+
+    private final Pattern includeFileName;
+    private final Pattern includePackageName;
+    private final Pattern includeModuleName;
+    private final Pattern includeCategory;
+    private final Pattern includeType;
+
+    private final Pattern excludeFileName;
+    private final Pattern excludePackageName;
+    private final Pattern excludeModuleName;
+    private final Pattern excludeCategory;
+    private final Pattern excludeType;
+
+    /**
+     * Ctor for Builder.
+     *
+     * @param builder
+     *         contains the filter-parameters.
+     */
+    private IssueFilterConfig(IssueFilterConfigBuilder builder) {
+        this.isIncludeSet = builder.isIncludePatternSet;
+        this.includeFileName = builder.includeFileName;
+        this.includePackageName = builder.includePackageName;
+        this.includeModuleName = builder.includeModuleName;
+        this.includeCategory = builder.includeCategory;
+        this.includeType = builder.includeType;
+        this.excludeFileName = builder.excludeFileName;
+        this.excludePackageName = builder.excludePackageName;
+        this.excludeModuleName = builder.excludeModuleName;
+        this.excludeCategory = builder.excludeCategory;
+        this.excludeType = builder.excludeType;
+    }
+
+    public boolean isIncludePatternSet() {
+        return isIncludeSet;
+    }
+
+    public Pattern getIncludeFileName() {
+        return includeFileName;
+    }
+
+    public Pattern getIncludePackageName() {
+        return includePackageName;
+    }
+
+    public Pattern getIncludeModuleName() {
+        return includeModuleName;
+    }
+
+    public Pattern getIncludeCategory() {
+        return includeCategory;
+    }
+
+    public Pattern getIncludeType() {
+        return includeType;
+    }
+
+    public Pattern getExcludeFileName() {
+        return excludeFileName;
+    }
+
+    public Pattern getExcludePackageName() {
+        return excludePackageName;
+    }
+
+    public Pattern getExcludeModuleName() {
+        return excludeModuleName;
+    }
+
+    public Pattern getExcludeCategory() {
+        return excludeCategory;
+    }
+
+    public Pattern getExcludeType() {
+        return excludeType;
+    }
+
+    /**
+     * A Builder for IssueFilterConfig. It initializes the includes and excludes with "matches nothing".
+     */
+    public static class IssueFilterConfigBuilder {
+        // is set to true, whenever a include pattern is set.
+        private boolean isIncludePatternSet = false;
+
+        // Patterns for include. Initialized with a Regex that matches nothing.
+        private Pattern includeFileName = Pattern.compile("(?!)");
```

Comment:
`Pattern.compile("(?!)")` auslagern in `private static final Pattern MATCHES_NOTHING` oder ähnliches (Pattern#compile durchaus aufwändig)